src/mesh/BoundingBox.cpp

```diff
@@ -13,50 +13,60 @@ namespace precice::mesh {
 
 logging::Logger BoundingBox::_log{"mesh::BoundingBox"};
 
+BoundingBox::BoundingBox(Eigen::VectorXd boundMin, Eigen::VectorXd boundMax)
+{
+  PRECICE_ASSERT(boundMin.rows() == boundMax.rows(), "Dimension of min {} and max {} vertices should be the same.", boundMin.rows(), boundMax.rows());
+  PRECICE_ASSERT((boundMin - boundMax).maxCoeff() < 0, "Each component of min vertex {} must be <= max vertex {} in the same axis direction.", boundMin, boundMax);
+
+  _boundMin   = std::move(boundMin);
+  _boundMax   = std::move(boundMax);
+  _dimensions = _boundMin.rows();
+}
+
 BoundingBox::BoundingBox(std::vector<double> bounds)
 {
-  PRECICE_ASSERT((int) bounds.size() == 4 || (int) bounds.size() == 6, "Dimension of a bounding box can only be 2 or 3.", bounds.size() / 2);
-  _bounds     = std::move(bounds);
-  _dimensions = _bounds.size() / 2;
+  PRECICE_ASSERT((int) bounds.size() == 4 || (int) bounds.size() == 6, "Dimension of a bounding box can only be 2 or 3. The dimension is {}.", bounds.size() / 2);
+  _dimensions = bounds.size() / 2;
+  _boundMin   = Eigen::Map<Eigen::VectorXd, 0, Eigen::InnerStride<2>>(bounds.data(), _dimensions);
+  _boundMax   = Eigen::Map<Eigen::VectorXd, 0, Eigen::InnerStride<2>>(bounds.data() + 1, _dimensions);
```

Comment:
Can you code comment what these lines do? Where is the `1` coming from? Is it really valid for all dimensions?

Reply:
1. Yes. One question please: should I code the comment also in hpp file or in the cpp file?

2. `1` is to move the pointer `bounds.array()` to point at its second element.

3. Yes, because the dimensions are only 2 and 3. This reminds us to add an assert in the constructor `BoundingBox(Eigen::VectorXd boundMin, Eigen::VectorXd boundMax)` to ensure `boundMin` and `boundMax` to have the same dimensions of 2 or 3.
```cpp
PRECICE_ASSERT((boundMin.rows() == 2 && boundMax.rows() == 2) || (boundMin.rows() == 3 && boundMax.rows() == 3),
                 "Dimension of min {} and max {} vertices should be the same and both 2 or 3.", boundMin.rows(), boundMax.rows());
```


- [Note for 2.](https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html)
`Eigen::Map` maps an existing array to a Eigen's matrix or vector. In this case, its first input is the existing array and its second input is the size of output `Eigen::VectorXd`. The existing array is from the vector `bounds` by using `bounds.array()` to return its array pointer. The `bounds.array()+1` means that the elements of output `Eigen::VectorXd` starts from the second element of the existing array `bounds.array()`.

Reply:
Here, directly in the cpp file, thanks!

Reply:
Done.